wdl2cwl/main.py

```diff
@@ -28,18 +28,23 @@
 # eval is not on the official list of reserved words, but it is a built-in function
 
 
-def convert(doc: str) -> cwl.CommandLineTool:
+def convert(doc: str) -> Dict[str, List[str]]:
     """Convert a WDL workflow, reading the file, into a CWL workflow Python object."""
     wdl_path = os.path.relpath(doc)
     doc_tree = WDL.load(wdl_path)
 
     parser = Converter()
 
-    tasks = []
+    dict_of_tasks = {"cwlVersion": "v1.2"}
+    dict_of_tasks["$graph"] = []  # type: ignore
+
     for task in doc_tree.tasks:
-        tasks.append(parser.load_wdl_objects(task))
+        cwl_task_object = parser.load_wdl_objects(task).save()
+        if len(doc_tree.tasks) == 1:
+            return cwl_task_object
+        dict_of_tasks["$graph"].append(cwl_task_object)  # type: ignore
 
-    return tasks[0]
+    return dict_of_tasks  # type: ignore
```

Comment:
@Th3nn3ss I haven't seen you using [list](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) & [dictionary](https://docs.python.org/3/tutorial/datastructures.html#dictionaries) comprehensions yet. That's something that took me some time getting used to in Python, but it has some performance benefits, besides being really common by Python devs. Here's a variation of this function with list comprehensions, and creating-and-returning the dictionary in one go (while, IMO, still keeping readability - it's important to remember readability for yourself & fellow devs)

```py
def convert(doc: str) -> Dict[str, Any]:
    """Convert a WDL workflow, reading the file, into a CWL workflow Python object."""
    wdl_path = os.path.relpath(doc)
    doc_tree = WDL.load(wdl_path)

    parser = Converter()

    if len(doc_tree.tasks) == 1:
        return parser.load_wdl_objects(doc_tree.tasks[0]).save()
    else:
        return {
            "cwlVersion": "v1.2",
            "$graph": [parser.load_wdl_objects(task).save() for task in doc_tree.tasks],
        }
```

p.s.: list/dict comprehension can also lead to performance [problems](https://realpython.com/list-comprehension-python/#when-not-to-use-a-list-comprehension-in-python), actually, and make the code harder to read & debug, so you don't need to replace every for-loop by a list comprehension :slightly_smiling_face:

Reply:
I replaced the return type, and removed the `type: ignore` markers, and `make mypy` passed for me locally with the code above. Maybe changing the return type will remove the ned to ignore types here @Th3nn3ss ? The list comprehension change is optional :+1: